packages/material-ui/src/Unstable_TrapFocus/Unstable_TrapFocus.js

```diff
@@ -141,10 +155,21 @@ function Unstable_TrapFocus(props) {
     };
   }, [disableAutoFocus, disableEnforceFocus, disableRestoreFocus, isEnabled, open]);
 
+  const onFocus = (event) => {
+    // detect focus inside children
+    insideReactTreeRef.current = true;
+    insideReactTreeEventTarget.current = event.target;
```

Comment:
Can we clean this up at some point? It seems like we can hold references to elements that have already been removed from the DOM. Since DOM elements point to their React Fibers we could potentially hold full  React trees that are already removed. No worries if we can't.

Reply:
Good point, I can at least clean it up when I am reseting the insideReactTree ref.